### DISCUSSION TITLE
Incomplete measures on musescore exports

### DISCUSSION THREAD

**Discussion by stevezemlicka** [Audiveris/audiveris#519] [Q&A]
First and foremost, what an amazing project!  I really wish I had come across this some time ago (before buying photoscore).  That being said, there are many detections and  other activities with which I believe Audiveris to be far superior and I really appreciate the efforts put into this.  I've been working with it for several weeks now and while the learning curve is a bit challenging, I think I've got the hang of much of the core functionality.  However I am still having difficulty getting usable musescore exports (though I'm getting pretty decent PDF exports).

On the sheets I've tried to do so far, I get numerous measure incomplete errors when importing to musescore.  I suspect it has something to do with staff connections but I think I've merged the ones that needed to be.  Would someone mind giving me a pointer to what I may be missing here?
[example.zip](https://github.com/Audiveris/audiveris/files/7521756/example.zip)

**Comment by hbitteur** [comment]
I just tried to reprocess your example.

On first sheet, some staff connections are missing because there is not enough margin on the left side of the image (a better centered scan would avoid this). Anyway, it's easy to manually connect the broken systems (via popup menu `system #n | Merge with system below`). No significant difficulty with the other sheets 2 and 3.

When exporting via MusicXML to Finale, I noticed something strange with the very first system of sheet 1, which appeared to be somehow "mixed" with the 2nd system. A null pointer exception occurred while visiting this system, which may have caused the export problem.

Moreover, when exporting the same data to MuseScore, I got a message about a corrupted file, with these details:

> Measure 31, staff 1 incomplete. Expected: 4/4; Found: 0/1
> Measure 32, staff 1 incomplete. Expected: 4/4; Found: 0/1
> Measure 33, staff 1 incomplete. Expected: 4/4; Found: 0/1

I'll need further investigation. But it's getting late and I will be away for a couple of days. Please be patient.

**Reply by stevezemlicka** [reply]
Take your time, this is a personal project so no rush.  I thought I had connected the systems but in hindsight might have messed with that as I was trying things out...you are correct about connecting the systems. What's interesting is i get the same measure incomplete error except for measures 4-29.  It's also interesting to note that this error is just for staff 1 on measures 4-10 but then starts for staffs 1 and 2 for measures 11-21 and then produces the error for staffs 1, 2, and 3 for measures 22-29.

I'll keep working on it and post if I find the culprit.  Thanks for checking it out.

**Comment by hbitteur** [comment]
Please find in 'development' branch a first bug fix (commit 89d37869bad090a4de76871d41b81f8e32591736).

It fixes the incomplete exported measures at the beginning of the score, because of dummy measures created on-the-fly for part 1 in first system. There was a bug in this creation, due to a recent modification on measure-long rests...

To be able to work immediately with the very latest development, you'll have to rebuild Audiveris (rather than waiting for a future release). See the [dedicated HandBook section](https://audiveris.github.io/audiveris/_pages/install/sources/). Make sure to checkout the development branch. If you are not familiar with this, don't hesitate to ask for help.

There is still a problem at the very end of this score (sheet 3), where alternate endings are not correctly handled. The culprit is the ending number 1, which starts in measure 29, located in a different system than ending number 2. It never came to my mind that endings could span systems breaks, so further and deeper work is still needed here. Stay tuned.
